### PR TITLE
replace RocksEngine with RaftEngine in cmd

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?
Replace `RocksEngine` with `RaftEngine` in cmd directory.

### Check List 

Tests 

- Unit test

Side effects

### Release note 
* No release note

### CODE REVIEW

File: cmd/src/server.rs [hunk 1]
```diff
@@ -90,66 +92,82 @@ pub fn run_tikv(config: TiKvConfig) {
...
+        let gc_worker = tikv.init_gc_worker();
+        let server_config = tikv.init_servers(&gc_worker);
+        tikv.register_services();
+        tikv.init_metrics_flusher();
+        tikv.run_server(server_config);
+        tikv.run_status_server();
+
+        signal_handler::wait_for_signal(Some(tikv.engines.take().unwrap().engines));
+        tikv.stop();
+    }
```
Comment (hunterlxt) [inline cmd/src/server.rs:129]: There are too many duplicate parts. It is better to use `Either` or wrap it with enum

File: cmd/src/server.rs [hunk 2]
```diff
@@ -915,6 +857,146 @@ impl TiKVServer {
...
+        let config_raftdb = &self.config.raftdb;
+        let mut raft_db_opts = config_raftdb.build_opt();
+        raft_db_opts.set_env(env.clone());
+        let raft_db_cf_opts = config_raftdb.build_cf_opts(&block_cache);
+        let raft_engine = engine_rocks::raw_util::new_engine_opt(
+            raft_db_path.to_str().unwrap(),
+            raft_db_opts,
+            raft_db_cf_opts,
+        )
+        .unwrap_or_else(|s| fatal!("failed to create raft engine: {}", s));
```
Comment (hunterlxt) [inline cmd/src/server.rs:876]: Same suggestion. It is better to use `Either` or wrap it with enum

File: etc/config-template.toml [hunk 3]
```diff
@@ -747,6 +747,12 @@
 # optimize-filters-for-hits = false
 
 [raftdb]
+## TiKV supports 2 type raft engines: `rocks` and `raft-engine`.
+## `rocks` is the traditional one, which is RocksDB exactly.
+## `raft-engine` is a new file-based implementation.
+type = "rocks"
+
+## For rocks.
```
Comment (hunterlxt) [inline etc/config-template.toml:755]: it's better to use subconfig here. For example:
```
[raftdb]
type = "rocks"
[raftdb.rocks]
...
[raftdb.raft-engine]
...
```

Comment (hicqu) [reply etc/config-template.toml:755]: It's old code. It can be removed.

File: src/config.rs [hunk 4]
```diff
@@ -2085,6 +2109,9 @@ pub struct TiKvConfig {
     #[config(submodule)]
     pub raftdb: RaftDbConfig,
 
+    #[config(skip)]
+    pub raft_engine: RaftEngineConfig,
```
Comment (NingLin-P) [inline src/config.rs:2113]: Should add a `Module` variant for `raft_engine` like `Module::RaftEngine`

File: src/config.rs [hunk 5]
```diff
@@ -2162,6 +2190,11 @@ impl TiKvConfig {
             self.raft_store.raftdb_path = config::canonicalize_path(&self.raft_store.raftdb_path)?;
         }
 
+        let default_er_path = config::canonicalize_sub_path(&self.storage.data_dir, "raft-engine")?;
+        if self.raft_engine.config.dir.is_empty() {
+            self.raft_engine.config.dir = default_er_path;
+        }
```
Comment (NingLin-P) [inline src/config.rs:2196]:
```suggestion
        let default_er_path = config::canonicalize_sub_path(&self.storage.data_dir, "raft-engine")?;
        if self.raft_engine.config.dir.is_empty() {
            self.raft_engine.config.dir = default_er_path;
        } else if self.raft_engine.config.dir != default_er_path {
            self.raft_engine.config.dir = config::canonicalize_path(&self.raft_engine.config.dir)?;
        }
```

Check out https://github.com/tikv/tikv/pull/6998#discussion_r390263181 https://github.com/tikv/tikv/pull/6998#discussion_r393707908

File: src/config.rs [hunk 6]
```diff
@@ -2162,6 +2190,11 @@ impl TiKvConfig {
...
+
```
Comment (NingLin-P) [inline src/config.rs:2197]: Should we also do some checks like https://github.com/tikv/tikv/blob/a562e0de60aa6e73b1796ad8d9a7b45287401044/src/config.rs#L2165-L2176 and https://github.com/tikv/tikv/blob/a562e0de60aa6e73b1796ad8d9a7b45287401044/src/config.rs#L2315

**Review by NingLin-P — APPROVED** [review]
LGTM

**Review by hunterlxt — APPROVED** [review]

**Pull request merged**